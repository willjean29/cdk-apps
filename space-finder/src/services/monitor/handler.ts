import { SNSEvent } from "aws-lambda";

const webhookUrl = "https://discord.com/api/webhooks/1288253235831771176/RfG04ozOhfvB5_ltUQ1IgXlf3j6IC49QnmNbTpZqB0PhV_29LKIYT6uPjxTgU2fX5p64";

export async function handler(event: SNSEvent, context: any) {
  const records = event.Records;
  for (const record of records) {
    try {
      await fetch(webhookUrl, {
        method: "POST",
        body: JSON.stringify({ content: `Huston, we have a problem: ${record.Sns.Message}` }),
        headers: {
          "Content-Type": "application/json",
        },
      });
    } catch (error) {
      console.log({ error });
    }
  }
}
