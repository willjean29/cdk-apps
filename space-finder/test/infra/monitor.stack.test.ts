import { App } from "aws-cdk-lib";
import { MonitorStack } from "../../src/infra/stacks/monitor.stack";
import { Template } from "aws-cdk-lib/assertions";

describe("Initial test suite ", () => {
  let monitorStackTemplate: Template;
  beforeEach(() => {
    const testApp = new App({
      outdir: "cdk.out",
    });
    const monitorStack = new MonitorStack(testApp, "MonitorStack");
    monitorStackTemplate = Template.fromStack(monitorStack);
  });

  it("lambda properties", () => {
    monitorStackTemplate.hasResourceProperties("AWS::Lambda::Function", {
      Handler: "index.handler",
      Runtime: "nodejs18.x",
    });
  });

  it("sns topic properties", () => {
    monitorStackTemplate.hasResourceProperties("AWS::SNS::Topic", {
      DisplayName: "AlarmTopic",
      TopicName: "AlarmTopic",
    });
  });
});
