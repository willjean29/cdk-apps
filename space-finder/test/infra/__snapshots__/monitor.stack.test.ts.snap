// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial test suite  Lambda stack snapshot 1`] = `
{
  "WebHookLambdaC2AD15FD": {
    "DependsOn": [
      "WebHookLambdaServiceRole1D0B8EE0",
    ],
    "Properties": {
      "Code": {
        "S3Bucket": {
          "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
        },
        "S3Key": "b99c592ff4db9f9d03e2e29bfe4c0ae155f76b53a9e4749af1f2deb51d82ee7c.zip",
      },
      "Handler": "index.handler",
      "Role": {
        "Fn::GetAtt": [
          "WebHookLambdaServiceRole1D0B8EE0",
          "Arn",
        ],
      },
      "Runtime": "nodejs18.x",
    },
    "Type": "AWS::Lambda::Function",
  },
}
`;

exports[`Initial test suite  Monitor stack snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AlarmTopicD01E77F9": {
      "Properties": {
        "DisplayName": "AlarmTopic",
        "TopicName": "AlarmTopic",
      },
      "Type": "AWS::SNS::Topic",
    },
    "SpacesApi4xxAlarm929EC379": {
      "Properties": {
        "AlarmActions": [
          {
            "Ref": "AlarmTopicD01E77F9",
          },
        ],
        "AlarmName": "SpacesApi4xxAlarm",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": [
          {
            "Name": "ApiName",
            "Value": "SpaceFinderApi",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "4XXError",
        "Namespace": "AWS/ApiGateway",
        "OKActions": [
          {
            "Ref": "AlarmTopicD01E77F9",
          },
        ],
        "Period": 60,
        "Statistic": "Sum",
        "Threshold": 5,
        "Unit": "Count",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "WebHookLambdaAlarmTopicD6696FB1": {
      "Properties": {
        "Endpoint": {
          "Fn::GetAtt": [
            "WebHookLambdaC2AD15FD",
            "Arn",
          ],
        },
        "Protocol": "lambda",
        "TopicArn": {
          "Ref": "AlarmTopicD01E77F9",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "WebHookLambdaAllowInvokeMonitorStackAlarmTopic9F999C82C57D735E": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "WebHookLambdaC2AD15FD",
            "Arn",
          ],
        },
        "Principal": "sns.amazonaws.com",
        "SourceArn": {
          "Ref": "AlarmTopicD01E77F9",
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "WebHookLambdaC2AD15FD": {
      "DependsOn": [
        "WebHookLambdaServiceRole1D0B8EE0",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "b99c592ff4db9f9d03e2e29bfe4c0ae155f76b53a9e4749af1f2deb51d82ee7c.zip",
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "WebHookLambdaServiceRole1D0B8EE0",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "WebHookLambdaServiceRole1D0B8EE0": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`Initial test suite  SnsTopic stack snapshot 1`] = `
{
  "AlarmTopicD01E77F9": {
    "Properties": {
      "DisplayName": "AlarmTopic",
      "TopicName": "AlarmTopic",
    },
    "Type": "AWS::SNS::Topic",
  },
}
`;
